echoregions/regions2d/regions2d.py

```diff
@@ -107,132 +110,128 @@ def select_region(
             and each row has time and depth within or on the boundaries passed
             in by the ``time_range`` and ``depth_range`` values.
         """
-        # Make copy of original dataframe; else, use original dataframe in selection.
-        if copy:
-            region = self.data.copy()
-        else:
-            region = self.data
+        # Make copy of the original dataframe; else, use the original dataframe in selection.
+        region = self.data.copy() if copy else self.data
+
+        # Check and subset for region_id
         if region_id is not None:
             if isinstance(region_id, (float, int, str)):
                 region_id = [region_id]
-            elif not isinstance(region_id, list):
-                raise TypeError(
-                    f"Invalid region_id type: {type(region_id)}. Must be \
-                                of type float, int, str, list, ``None``."
-                )
-            # Select row by column id
-            for value in region_id:
-                if not isinstance(value, (float, int, str)):
-                    raise TypeError(
-                        f"Invalid element in list region_id. Is of \
-                            type: {type(value)}Must be \
-                            of type float, int, str."
-                    )
-            region = self.data[self.data["region_id"].isin(region_id)]
-        if time_range is not None:
-            if isinstance(time_range, List):
-                if len(time_range) == 2:
-                    if isinstance(time_range[0], Timestamp) and isinstance(
-                        time_range[1], Timestamp
-                    ):
-                        if time_range[0] < time_range[1]:
-                            # Select rows with time values that are all within time range
-                            region = region[
-                                region["time"].apply(
-                                    lambda time_array: all(
-                                        time_range[0] <= Timestamp(x)
-                                        and time_range[1] >= Timestamp(x)
-                                        for x in time_array
-                                    )
-                                )
-                            ]
-                        else:
-                            raise ValueError(
-                                f"1st index value must be later than 0th index \
-                                             value. Currently 0th index value is {time_range[0]} \
-                                             and 1st index value is {time_range[1]}"
-                            )
-                    else:
+            elif isinstance(region_id, list):
+                for value in region_id:
+                    if not isinstance(value, (float, int, str)):
```

Comment:
seems like the list comprehension you have below for the timestamp part looks more succinct?
```
all(isinstance(t, Timestamp) for t in time_range)
```

Reply:
Oh actually I see now that you use a for loop to print the exact type out. I'll resolve this comment.